shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -10,165 +10,112 @@
-                                }).collect(Collectors.toList())));
-    }
-
-    @Override
-    public Uni<Object> fetchAndProcessProcessorsToDeployOrDelete() {
-        return getAuthenticatedRequest(webClientManager.get(APIConstants.SHARD_API_BASE_PATH + "processors"), HttpRequest::send)
-                .onItem().invoke(success -> updateManagerRequestMetricsOnSuccess(ManagerRequestType.FETCH, success))
-                .onFailure().invoke(failure -> updateManagerRequestMetricsOnFailure(ManagerRequestType.FETCH, failure))
-                .onItem().transform(this::getProcessors)
-                .onItem().transformToUni(x -> Uni.createFrom().item(x.stream()
-                        .map(y -> {
-                            if (ManagedResourceStatus.ACCEPTED.equals(y.getStatus())) {
-                                y.setStatus(ManagedResourceStatus.PROVISIONING);
-                                return notifyProcessorStatusChange(y)
-                                        .subscribe().with(
-                                                success -> {
-                                                    LOGGER.debug("Provisioning notification for Processor '{}' has been sent to the manager successfully", y.getId());
-                                                    bridgeExecutorService.createBridgeExecutor(y);
-                                                },
-                                                failure -> failedToSendUpdateToManager(y, failure));
-                            }
-                            if (ManagedResourceStatus.DEPROVISION.equals(y.getStatus())) { // Processor to delete
-                                y.setStatus(ManagedResourceStatus.DELETING);
-                                return notifyProcessorStatusChange(y)
-                                        .subscribe().with(
-                                                success -> {
-                                                    LOGGER.debug("Deleting notification for Processor '{}' has been sent to the manager successfully", y.getId());
-                                                    bridgeExecutorService.deleteBridgeExecutor(y);
-                                                },
-                                                failure -> failedToSendUpdateToManager(y, failure));
-                            }
-                            return Uni.createFrom().voidItem();
-                        }).collect(Collectors.toList())));
-    }
-
-    private List<ProcessorDTO> getProcessors(HttpResponse<Buffer> httpResponse) {
-        return deserializeResponseBody(httpResponse, new TypeReference<List<ProcessorDTO>>() {
-        });
-    }
-
-    private List<BridgeDTO> getBridges(HttpResponse<Buffer> httpResponse) {
-        return deserializeResponseBody(httpResponse, new TypeReference<List<BridgeDTO>>() {
-        });
+    protected class BridgeHandler implements Function<List<BridgeDTO>, Uni<Object>> {
```

Comment:
The `ManagerClient` only handles HTTP interaction with the `manager`.

Processing of the responses is still handled in the `ManagerSyncService`.